### PR TITLE
Fix payment method type

### PR DESCRIPTION
**Pull request by znarf**
Frontend: https://github.com/opencollective/opencollective-frontend/pull/6760

This was a rabbit hole but I think that PR put us in the right direction to clean that technical debt.

### CODE REVIEW

File: server/graphql/v2/enum/PaymentMethodType.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 
-import { PAYMENT_METHOD_TYPES } from '../../../constants/paymentMethods';
+import { PAYMENT_METHOD_TYPE } from '../../../constants/paymentMethods';
 
 export const PaymentMethodType = new GraphQLEnumType({
   name: 'PaymentMethodType',
-  values: PAYMENT_METHOD_TYPES.reduce((values, key) => {
-    return { ...values, [key]: {} };
+  values: Object.keys(PAYMENT_METHOD_TYPE).reduce((values, key) => {
+    return { ...values, [key]: { value: PAYMENT_METHOD_TYPE[key] } };
```
Comment (Betree) [inline server/graphql/v2/enum/PaymentMethodType.ts:8]: Too keep it retro-compatible, I would suggest:

```suggestion  values: {
      ...PAYMENT_METHOD_TYPES.reduce(
      (values, key) => ({ ...values, [key]: { deprecationReason: 'Please use uppercase values' } }),
      {},
    ),
    ...Object.keys(PAYMENT_METHOD_TYPE).reduce(
      (values, key) => ({ ...values, [key]: { value: PAYMENT_METHOD_TYPE[key] } }),
      {},
    ),
```

Comment (znarf) [reply server/graphql/v2/enum/PaymentMethodType.ts]: Thank you for the tip, trying this.

**Comment by znarf** [comment]
@Betree E2E tests are still failing, looking at it.

File: server/constants/paymentMethods.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export enum PAYMENT_METHOD_TYPE {
   COLLECTIVE = 'collective',
   HOST = 'host',
   ADAPTIVE = 'adaptive',
-  GIFT_CARD = 'giftcard',
```
Comment (Betree) [inline server/constants/paymentMethods.ts:20]: I would tend to the opposite direction: all entries should have this`_` (eg. `CREDIT_CARD`) as that's how things are formatted in our other enums (snake uppercase).

Comment (znarf) [reply server/constants/paymentMethods.ts:20]: I'm favoring consistency with the database values for now.

I get that constants are effectively here to help having nicer/different values than the database one.

But as we already suffered too much on this topic, I think we need to relax a bit with a simple mental model.

**Pull request merged**